tools/packages/wesl/src/BindIdents.ts

```diff
@@ -160,19 +169,29 @@ function bindIdentsRecursive(
   return [newDecls, newFromChildren, newFromRefs].flat();
 }
 
-function setDisplayName(
+/**
+ * Mutate a DeclIdent to set a unique name for global linking,
+ * using a mangling function to choose a unique name.
+ * Also update the set of globally unique names.
+ */
+function setMangledName(
   proposedName: string,
   decl: DeclIdent,
   globalNames: Set<string>,
+  srcModule: SrcModule,
+  mangler: ManglerFn,
 ): void {
   if (!decl.mangledName) {
+    let mangledName: string;
     if (decl.declElem && isGlobal(decl.declElem)) {
       const sep = proposedName.lastIndexOf("::");
       const name = sep === -1 ? proposedName : proposedName.slice(sep + 2);
-      decl.mangledName = declUniqueName(name, globalNames);
+      mangledName = mangler(decl, srcModule, name, globalNames);
```

Comment:
Very neat, making `mangler` a function.

I suspect that at some point in the future, it'll be something like
```
interface Mangler {
  startScope();
  mangle(decl, srcModule, name);
  endScope();
}
```

because then the mangler could do the necessary scope tracking to emit the shortest possible names.

But anyways, that's a far future project.